core/arch/arm/include/kernel/spinlock.h

```diff
@@ -69,7 +69,7 @@ static inline unsigned int cpu_spin_trylock(unsigned int *lock)
 
 	assert(thread_irq_disabled());
 	locked = __cpu_spin_trylock(lock);
-	if (locked)
+	if (!locked)
```

Comment:
Yes, please.
Well, take a look in `spin_lock_a64.S` : `__cpu_spin_trylock()`. What's in `x0` when it returns?